tools/RestfulApiWebService/RestfulApiWebService/Controllers/QueryController.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.DocAsCode.Build.Engine;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Description;
+using System.Data.Entity;
+
+using System.Web.Script.Serialization;
+using Microsoft.DocAsCode.Plugins;
+using DB.Models;
+using System.Diagnostics;
+using DB.Dao;
+
+namespace RestfulApiService.Controllers
+{
+    [RoutePrefix("uids")]
+    public class QueryController : ApiController
+    {
+        private XrefSpecDBContext db = new XrefSpecDBContext();
+
+        [HttpGet]
+        [Route("{uid}")]
+        public async Task<IHttpActionResult> GetByUid(string uid)
+        {
+            var ut = await db.uidTable.Where(b => b.uid == uid)
+                                .Select(c => c.objectStr)
+                                .ToListAsync();
+           
+            List<XRefSpec> xf = Newtonsoft.Json.JsonConvert.DeserializeObject<List<XRefSpec>>("[" + string.Join(",", ut) + "]");
```

Comment:
[https://stackoverflow.com/questions/9541351/returning-binary-file-from-controller-in-asp-net-web-api](https://stackoverflow.com/questions/9541351/returning-binary-file-from-controller-in-asp-net-web-api)